### PR TITLE
[Contacts list] Add the option to search in the users directory

### PR DESCRIPTION
**Pull request by yostyle**
#84

### CODE REVIEW

**Comment by yostyle** [comment]
> @yostyle can you please replace the blue background color of the button with white color
> image

**Comment by yostyle** [comment]
> @yostyle can you please replace the blue background color of the button with white color
> image

This background color has been updated

**Review by giomfo — APPROVED** [review]

File: vector/src/main/java/im/vector/app/features/home/HomeDetailFragment.kt [hunk 1]
```diff
@@ -188,6 +191,14 @@ class HomeDetailFragment @Inject constructor(
                 view.doOnNextLayout { closeSearchView() }
             }
         }
+
+        tchapContactListViewModel.observeViewEvents {
+            when (it) {
+                is TchapContactListViewEvents.OpenSearch -> if (!views.homeSearchView.isVisible) toggleSearchView()
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/home/HomeDetailFragment.kt:197]: I think we can a dedicated method "openSearchView"  to remove this "trick"

File: vector/src/main/java/im/vector/app/features/home/HomeDetailFragment.kt [hunk 2]
```diff
@@ -188,6 +191,14 @@ class HomeDetailFragment @Inject constructor(
...
+                // TODO view.doOnNextLayout { closeSearchView() }
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/home/HomeDetailFragment.kt:198]: remove or detail this todo

File: vector/src/main/java/fr/gouv/tchap/features/home/contact/list/TchapContactListViewState.kt [hunk 3]
```diff
@@ -40,9 +40,14 @@ data class TchapContactListViewState(
...
+
+        private val showSearch: Boolean
 ) : MvRxState {
     constructor(args: TchapContactListFragmentArgs) : this(
-            excludedUserIds = args.excludedUserIds
+            excludedUserIds = args.excludedUserIds,
+            showSearch = args.showSearch
     )
+
+    fun showSearch() = showSearch
```
Comment (Florian14) [inline vector/src/main/java/fr/gouv/tchap/features/home/contact/list/TchapContactListViewState.kt:52]: I don't understand the goal of this method and why the attribute is private ?

File: vector/src/main/java/im/vector/app/features/home/HomeDetailFragment.kt [hunk 4]
```diff
@@ -194,8 +194,9 @@ class HomeDetailFragment @Inject constructor(
 
         tchapContactListViewModel.observeViewEvents {
             when (it) {
-                is TchapContactListViewEvents.OpenSearch -> if (!views.homeSearchView.isVisible) toggleSearchView()
-                // TODO view.doOnNextLayout { closeSearchView() }
+                is TchapContactListViewEvents.OpenSearch   -> if (!views.homeSearchView.isVisible) openSearchView()
```
Comment (Florian14) [inline vector/src/main/java/im/vector/app/features/home/HomeDetailFragment.kt:197]: I'm not sure this condition is still necessary

**Pull request merged**